design-proposals/network-binding-plugin/network-binding-plugin.md

```diff
@@ -941,3 +1001,154 @@ Libvirt configuration snippet:
     <rom enabled='no'/>
 </interface>
 ```
+
+## Appendix F: Concrete example for vDPA binding
+
+Present a concrete example of the actions needed at each integration
+point, considering the option suggested in the
+[design section](#design-details).
+
+### Kubevirt CR Specification
+
+The vDPA binding is registered to the Kubevirt CR.
+The plugin is then referenced by name from the VM Specification.
+
+> **Note**: The vDPA binding does not need to perform changes in the
+> pod network, therefore, there is no need for a
+> network-attachment-definition.
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  configuration:
+    network:
+      binding:
+        vdpa:
+          sidecarImage: quay.io/kubevirt/network-vdpa-binding
+          downwardAPI: device-info
```

Comment:
implementation wise, should this be a list of strings? do you anticipate that there could be more than one metadata information that could be passed to sidecar network plugins?

Reply:
I think that in the future there might be other data that a plugin would require but I don't think that a combination of more than one type will be required. And even if it would be (unlikely), we can always introduce a new type that represents a combination.

Reply:
As an example in the future consider that the network sidecar needs  `kubevirt.io/foo` annotation along with `kubevirt.io/network-info` as metadata in sidecar. How would this API evolve?

Reply:
`downwardAPI: device-info-foo` or comma separated `downwardAPI: deviceinfo,foo`. Or we can introduce `downwardAPIs`. Although I don't think this kind of combination will ever be required. I cannot think of a plugin that will need information from several pod fields.  I don't even believe that anything besides deviceinfo will be required, but kept this option open.

Reply:
This reminds me of K8S Pod status API where the IPS  filed was added once IP was not sufficient.
IMO it is not nice to add the another filed just to support multiple options, I would go with sting slice or support coma separated filed in the future (not optimal).